src/renderer/api-decorator.js

```diff
@@ -524,18 +524,37 @@ limitations under the License.
      * Preload script eval
      */
     ipc.once(`post-api-injection-${renderFrameId}`, () => {
-        const preload = windowOptions.preload;
+        const preloads = windowOptions.preload;
 
-        if (preload) {
-            try {
-                const preloadScriptContent = syncApiCall('get-window-preload-script', {
-                    preload
-                });
-                window.eval(preloadScriptContent); /* jshint ignore:line */
-            } catch (err) {
-                console.error(err);
-            }
+        // may be a string primitive OR an array; must have length in any case
+        if (!(preloads && preloads.length)) {
+            return;
+        }
+
+        try {
+            fin.preloads = syncApiCall('get-window-preload-scripts', { preloads });
```

Comment:
We already know this will need to be split up into:

Browser side: Download scripts
API Decorator: evaluate them